Example fails on macOS Big Sur (11.0.1)

**Describe the bug**
rules_haskell fails on newest macOS because of `sed` issue. Other downstream issues may exist.

**To Reproduce**
```bash
curl https://haskell.build/start | sh
bazel build //...
```

```
Starting local Bazel server and connecting to it...
INFO: Repository rules_haskell_ghc_darwin_amd64 instantiated at:
  no stack (--record_rule_instantiation_callstack not enabled)
Repository rule _ghc_bindist defined at:
  /private/var/tmp/_bazel_liam/8a2700302396c489f271b3938aaa9254/external/rules_haskell/haskell/ghc_bindist.bzl:362:31: in <toplevel>
ERROR: An error occurred during the fetch of repository 'rules_haskell_ghc_darwin_amd64':
   Traceback (most recent call last):
	File "/private/var/tmp/_bazel_liam/8a2700302396c489f271b3938aaa9254/external/rules_haskell/haskell/ghc_bindist.bzl", line 292
		execute_or_fail_loudly(ctx, <1 more arguments>)
	File "/private/var/tmp/_bazel_liam/8a2700302396c489f271b3938aaa9254/external/rules_haskell/haskell/private/workspace_utils.bzl", line 18, in execute_or_fail_loudly
		fail(<1 more arguments>)
Command failed: ./patch_bins
sed: bin/ghc: in-place editing only works for regular files
sed: bin/ghc-pkg: in-place editing only works for regular files
sed: bin/ghci: in-place editing only works for regular files
sed: bin/haddock: in-place editing only works for regular files
sed: bin/runghc: in-place editing only works for regular files
sed: bin/runhaskell: in-place editing only works for regular files
ERROR: /Users/liam/Work/bazel_test/BUILD.bazel:15:26: //:base depends on @rules_haskell_ghc_darwin_amd64//:toolchain-impl in repository @rules_haskell_ghc_darwin_amd64 which failed to fetch. no such package '@rules_haskell_ghc_darwin_amd64//': Traceback (most recent call last):
	File "/private/var/tmp/_bazel_liam/8a2700302396c489f271b3938aaa9254/external/rules_haskell/haskell/ghc_bindist.bzl", line 292
		execute_or_fail_loudly(ctx, <1 more arguments>)
	File "/private/var/tmp/_bazel_liam/8a2700302396c489f271b3938aaa9254/external/rules_haskell/haskell/private/workspace_utils.bzl", line 18, in execute_or_fail_loudly
		fail(<1 more arguments>)
Command failed: ./patch_bins
sed: bin/ghc: in-place editing only works for regular files
sed: bin/ghc-pkg: in-place editing only works for regular files
sed: bin/ghci: in-place editing only works for regular files
sed: bin/haddock: in-place editing only works for regular files
sed: bin/runghc: in-place editing only works for regular files
sed: bin/runhaskell: in-place editing only works for regular files
ERROR: Analysis of target '//:base' failed; build aborted: Analysis failed
INFO: Elapsed time: 84.056s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (19 packages loaded, 27 targets configured)
```

**Expected behavior**
Example builds

**Environment**
 - OS name + version: macOS Big Sur version 11.0.1
 - Bazel version: bazel 3.3.1
 - Version of the rules: v0.13

**Additional context**
* There are likely additional issues when using Big Sur.
* HEAD of master also has the same issue
```patch
- http_archive(
-     name="rules_haskell",
-     strip_prefix="rules_haskell-0.13",
-     urls=["https://github.com/tweag/rules_haskell/archive/v0.13.tar.gz"],
-     sha256 = "b4e2c00da9bc6668fa0404275fecfdb31beb700abdba0e029e74cacc388d94d6",
- )
+ load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
+ git_repository(
+     name="rules_haskell",
+     remote="git@github.com:tweag/rules_haskell.git",
+     commit="9a6b89928a4f6171743798b8d7792b54bd16c5da",
+     # shallow_since="1592537100 -0700",
+ )
```

I was testing this idea out as well and it gets me further along the build 👍 . I will validate #1445  (since it's slightly different from what I had).

Unfortunately, I was seeing more downstream issues, potentially unrelated to rules_haskell since they are in the zlib compilation part, that I was trying to fix via configuration. The TL;DR version is
```
external/zlib.dev/gzlib.c:252:9: error: implicit declaration of function 'lseek' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
```

--- a/haskell/ghc_bindist.bzl
+++ b/haskell/ghc_bindist.bzl
@@ -281,10 +281,11 @@ def _ghc_bindist_impl(ctx):
         execute_or_fail_loudly(ctx, ["./configure", "--prefix", bindist_dir.realpath])
         execute_or_fail_loudly(ctx, ["make", "install"])
         ctx.file("patch_bins", executable = True, content = r"""#!/usr/bin/env bash
-grep --files-with-matches --null {bindist_dir} bin/* | xargs -0 -n1 \
+find bin -type f -print0 | xargs -0 \
+grep --files-with-matches --null {bindist_dir} | xargs -0 -n1 \
     sed -i.bak \
         -e '2i\
-          DISTDIR="$( dirname "$(resolved="$0"; while tmp="$(readlink "$resolved")"; do resolved="$tmp"; done; echo "$resolved")" )/.."' \
+DISTDIR="$( dirname "$(resolved="$0"; cd "$(dirname "$resolved")"; while tmp="$(readlink "$(basename "$resolved")")"; do resolved="$tmp"; cd "$(dirname "$resolved")"; done; echo "$PWD/$(basename "$resolved")")" )/.."' \
         -e 's:{bindist_dir}:$DISTDIR:'
 """.format(
             bindist_dir = bindist_dir.realpath,

